packages/driver/src/cypress/log.ts

```diff
@@ -349,33 +349,64 @@ export class Log {
       next: null,
     })
 
-    const snapshot = this.cy.createSnapshot(name, this.get('$el'))
+    let snapshot
+    const postProcessSnapshot = (shouldRebindSnapshotFn = true) => {
+      const snapshots = this.get('snapshots') || []
+
+      // don't add snapshot if we couldn't create one, which can happen
+      // if the snapshotting process errors
+      // https://github.com/cypress-io/cypress/issues/15816
+      if (snapshot) {
+      // insert at index 'at' or whatever is the next position
+        snapshots[options.at || snapshots.length] = snapshot
+      }
 
-    const snapshots = this.get('snapshots') || []
+      this.set('snapshots', snapshots)
 
-    // don't add snapshot if we couldn't create one, which can happen
-    // if the snapshotting process errors
-    // https://github.com/cypress-io/cypress/issues/15816
-    if (snapshot) {
-      // insert at index 'at' or whatever is the next position
-      snapshots[options.at || snapshots.length] = snapshot
+      if (options.next && shouldRebindSnapshotFn) {
+        const fn = this.snapshot
+
+        this.snapshot = function () {
+        // restore the fn
+          this.snapshot = fn
+
+          // call orig fn with next as name
+          return fn.call(this, options.next)
+        }
+      }
+
+      return this
     }
 
-    this.set('snapshots', snapshots)
+    if (this.config('experimentalSessionAndOrigin') && !Cypress.isCrossOriginSpecBridge) {
+      // @ts-ignore
+      const { originPolicy: activeSpecBridgeOriginPolicyIfApplicable } = this.state('currentActiveOriginPolicy') ? Cypress.Location.create(this.state('currentActiveOriginPolicy')) : {
+        originPolicy: undefined,
+      }
+      // @ts-ignore
+      const { originPolicy: originPolicyThatIsSoonToBeOrIsActive } = Cypress.Location.create(this.state('anticipatingCrossOriginResponse')?.href || this.state('url'))
 
-    if (options.next) {
-      const fn = this.snapshot
+      if (activeSpecBridgeOriginPolicyIfApplicable && activeSpecBridgeOriginPolicyIfApplicable === originPolicyThatIsSoonToBeOrIsActive) {
```

Comment:
In what scenarios is this not true? If we are in a cy.origin block but have begun navigating to another origin? Does it make a difference if it's the top origin or another cross origin?

Reply:
Mainly the error case/operator error. If a user navigates to an origin that doesn't match the spec bridge origin, like 

```js
cy.get('[href="https://www.foobar.com"]').click()
cy.origin('https://www.barbaz.com', () => {
   // wrong origin, don't snapshot this
})
```


I added a test [here](https://github.com/cypress-io/cypress/pull/21552/files#diff-60e1b7d5d5f2e76025dc31ee733698210d305fa02a1a1a9dfa600728fb8a7170R83) to make sure this fails gracefully.